src/tool_getparam.c

```diff
@@ -541,24 +541,24 @@ static ParameterError GetSizeParameter(struct GlobalConfig *global,
   return PARAM_OK;
 }
 
-static void cleanarg(char *str)
-{
 #ifdef HAVE_WRITABLE_ARGV
+static void cleanarg(argv_item_t str)
+{
   /* now that GetStr has copied the contents of nextarg, wipe the next
    * argument out so that the username:password isn't displayed in the
    * system process list */
   if(str) {
-    size_t len = strlen(str);
+    size_t len = strlen((char *)str);
```

Comment:
Doesn't this only work coincidentally because `HAVE_WRITABLE_ARGV` is never set when `wchar_t` is used and otherwise the terminating null character would be found too early for ASCII characters?
Maybe just leave out the cast and change the signature back to `char *` if we can safely assume that this function is never used for `wchar_t`, or otherwise use `wcslen` in a `wchar_t` overload.

Reply:
I see your point.

But with that fix, we would instead get a warning because we pass in an `argv_item_t` to a `char *` parameter, so it would still need a typecast somewhere.

Reply:
Do any of the builds have strong type checking? The only other type at issue here is VMS __char_ptr32. Removing the cast could help us out with a warning if something like this happens again.